src/index.ts

```diff
@@ -140,14 +142,40 @@ class CommandBuilder {
   public parser: Argv.Parser
   public context!: Argv.Context
 
-  public constructor(parser: Argv.Parser, argv: string[], commandClasses: { new (): Command }[]) {
+  public constructor(parser: Argv.Parser) {
     this.parser = parser
     this.initedCommands = []
+  }
+
+  private autocomplete(options: ICli): Promise<void> {
+    return new Promise(resolve => {
+      const applicationName = options.application?.command || 'undefined'
+      const completion = omelette(applicationName)
+
+      completion.on('complete', (fragment, { line, reply }) => {
+        const relevantPart = line.slice(applicationName.length + 1)
+        reply(this.parser.suggest(relevantPart))
```

Comment:
won't this reply remove the initial command name?

Reply:
Yes, command name is irrelevant.